src/control.rs

```diff
@@ -242,32 +348,74 @@ impl ControlPoint {
         })
     }
 
-    pub async fn send_command(&self, device: &Peripheral, data: Vec<u8>) -> PolarResult<ControlResponse> {
-        println!("Writing cmd: {:?}", data);
+    /// Send command to Control Point
+    pub async fn send_command(&self, device: &Peripheral, data: Vec<u8>) -> PolarResult<()> {
         self.write(device, data).await?;
 
-        let response_data = self.read(device).await?;
-        println!("Read response: {:?}", response_data);
-        let mut response = ControlResponse::new(response_data).await?;
-
-        while response.more {
-            let response_data = self.read(device).await?;
-            println!("Read response (more): {:?}", response_data);
-            response.add_parameters(response_data);
-        }
-
-        Ok(response)
+        Ok(())
     }
 
     async fn write(&self, device: &Peripheral, data: Vec<u8>) -> PolarResult<()> {
-        device.write(&self.control_point, &data, WriteType::WithResponse)
+        device
+            .write(&self.control_point, &data, WriteType::WithResponse)
             .await
             .map_err(Error::BleError)
     }
 
-    async fn read(&self, device: &Peripheral) -> PolarResult<Vec<u8>> {
-        device.read(&self.control_point)
+    /// Read data from control point (for reading the features of a device)
+    pub async fn read(&self, device: &Peripheral) -> PolarResult<Vec<u8>> {
+        device
+            .read(&self.control_point)
             .await
             .map_err(Error::BleError)
     }
-}
\ No newline at end of file
+}
+
+#[cfg(test)]
+mod test {
+    use super::*;
+
+    // for async testing
+    macro_rules! aw {
+        ($e:expr) => {
+            tokio_test::block_on($e)
+        };
+    }
+
+    #[test]
+    fn settings_ecg() {
+        let norm = StreamSettings {
+            ty: H10MeasurementType::Ecg,
+            resolution: 14,
+            range: None,
+            sample_rate: vec![130],
+        };
+
+        let data = aw!(ControlResponse::new(vec![
+            0xf0, 0x01, 0x00, 0x00, 0x00, 0x00, 0x01, 0x82, 0x00, 0x01, 0x01, 0x0e, 0x00
+        ]))
+        .unwrap();
+        println!("data: {:?}", data);
```

Comment:
Do you need to print this data in the test?

Reply:
Not really, I was using it when it didn't work to see what the output was, when it wasn't working.